datavault-common/src/main/java/org/datavaultplatform/common/model/dao/RetrieveDAOImpl.java

```diff
@@ -93,4 +107,36 @@ public int inProgressCount() {
         session.close();
         return retrieves;
     }
+
+    private Optional<Criteria> retrieveCriteriaForUser(String userId, Session session, Permission permission) {
```

Comment:
Seeing as this is repeated across numerous DAOs (i.e. more than one), should that be separated and kept in a shared location (base class, helper class, wherever)? This ties to my comment above about disentangling those two rather distinct concepts: getting the schoolIds user has permission in and constructing the entity-specific criteria